tests/functional/src/main/java/org/mule/functional/classloading/isolation/builder/IsolatedClassLoaderExtensionsManagerConfigurationBuilder.java

```diff
@@ -70,23 +70,17 @@ protected void doConfigure(final MuleContext muleContext) throws Exception
     {
         final ExtensionManagerAdapter extensionManager = createExtensionManager(muleContext);
 
-        for (Object pluginClassLoader : pluginsClassLoaders)
+        for (ClassLoader pluginClassLoader : pluginsClassLoaders)
         {
-            String artifactName = (String) pluginClassLoader.getClass().getMethod("getArtifactName").invoke(pluginClassLoader);
-            ClassLoader classLoader = (ClassLoader) pluginClassLoader.getClass().getMethod("getClassLoader").invoke(pluginClassLoader);
-            URL manifestUrl = getExtensionManifest(classLoader);
+            URL manifestUrl = getExtensionManifest(pluginClassLoader);
             if (manifestUrl != null)
             {
-                if (LOGGER.isDebugEnabled())
-                {
-                    LOGGER.debug("Discovered extension: {}", artifactName);
-                }
                 ExtensionManifest extensionManifest = extensionManager.parseExtensionManifestXml(manifestUrl);
-                extensionManager.registerExtension(extensionManifest, classLoader);
+                extensionManager.registerExtension(extensionManifest, pluginClassLoader);
             }
             else
             {
-                LOGGER.debug("Discarding plugin artifact class loader with artifactName '{}' due to it doesn't have an extension descriptor", artifactName);
+                LOGGER.debug("Discarding plugin artifact class loader '{}' due to it doesn't have an extension descriptor", pluginClassLoader);
```

Comment:
This log won't be useful if the classloader does not provides a meaningful toString message